[DO NOT MERGE] proposed schema

kgroble opened the pull request:
I realized before I get any farther I should get some feedback on the schema I came up with. This drops a bunch of columns that used to exist whose purpose I couldn't figure out, but I wouldn't mind adding them back if there's a good reason to.

Highlights:
* Lots more tables, actual relations in our relational database
* It seemed to make more sense to track times as floats offset from 0 (the start of a game) rather than timestamps, since I'm guessing any analysis would just convert them back to 0-offset floats anyway
* gestures, commands, moves, and flips are all separate entities with their own tables...
* ... but the move_data and flip_data join tables exist so that we can bind sets of them together in AI mode and make it easy to go back and collect training data for the neural networks.

kgroble committed:
proposed schema

kgroble committed:
add start timestamp

Lewis-Kelley commented on server/create-baseline.sql:
```diff
@@ -1,25 +1,15 @@
 CREATE TABLE human_survey (
+  game_id int REFERENCES game(id),
+  commander_role boolean,
+  worker_role boolean,
```
You can go ahead and get rid of all references to the different kinds of roles. We're only making a single-player game now.

Lewis-Kelley commented on server/create-baseline.sql:
```diff
@@ -29,7 +19,57 @@ CREATE TABLE human_survey (
 );
 
 CREATE TABLE robot_survey (
```
We might not need this anymore either, since there's only one person taking the survey.

kgroble commented:
![new-schema-diagram](https://user-images.githubusercontent.com/6350161/38040869-72e9ae50-327e-11e8-8f26-64eab2d9623c.png)

Diagram of proposed schema.

Lewis-Kelley reviewed (changes requested):
There are a few things we don't need to store any more (most of them deal with coop), so those can be safely removed.

One thing I want to point out is that we don't store the initial setup in the database at all. Therefore, we cannot know what the state of the board was when the user made a command we failed to recognize.

kgroble committed:
added blocks table, cleaned up surveys

kgroble commented:
Added blocks table, removed now redundant color/letter information from flip table, added coordinates to flips (to help with things like "flip the bottom block"), added redo needed flags to data tables, consolidated survey tables.

kgroble commented:
![schema-with-blocks](https://user-images.githubusercontent.com/6350161/38099444-9c3ec61c-3348-11e8-9330-9b0020763324.png)

New diagram

Lewis-Kelley commented:
It looks like you didn't set any of the foreign `block_id`'s to be foreign keys.

kgroble commented:
It doesn't seem like it gains us much. The `block` table should have the full cross product of `game_id`s and valid `block_id`s in it, so does a foreign key from `(game_id, block_id)` to the `block` table do anything the foreign key on `game_id` to the `game` table doesn't?

Lewis-Kelley commented:
The functionality's the same, but the constraints are not. We can leave it as-is, since of all the data to get messed up, `block_id` is not one I'm really worried about.

rbubulka commented:
I disagree block id's should be the data were most concerned about getting messed up, in the event that our block ids get borked in the table somehow it will screw up the network since then we cant get a valid set of colors and letters to pair it with

kgroble committed:
added foreign keys to block_id for move and flip

kgroble commented:
![new-schema](https://user-images.githubusercontent.com/6350161/38288235-73d59bb6-379d-11e8-8b5c-32072557491e.png)

@rbubulka I added the constraints Lewis originally brought up, do you think there's any more we can do?
